sdk/inc/azure/core/internal/az_precondition_internal.h

```diff
@@ -117,8 +117,8 @@ AZ_NODISCARD AZ_INLINE bool _az_span_is_valid(az_span span, int32_t min_size, bo
   // left until the theoretical end of the address space, it is not a valid span.
   // Example: (az_span) { .ptr = (uint8_t*)(~0 - 5), .size = 10 } is not a valid span, because most
   // likely you end up pointing to 0x0000 at .ptr[6], &.ptr[7] is 0x...0001, etc.
-  uint8_t* const max_ptr = (uint8_t*)~0;
-  result &= ((size_t)span_size <= (size_t)(max_ptr - ptr));
+  uint8_t* const max_ptr = (uint8_t*)-1;
+  result = result && ((size_t)span_size <= (size_t)(max_ptr - ptr));
```

Comment:
Why did you change from a bitwise & to a logical &&?

Reply:
warning: use of a signed integer operand with a binary bitwise operator [hicpp-signed-bitwise]

Also https://stackoverflow.com/a/24560

Reply:
ah, yes I see those are both booleans now.  definitely more deterministic now. thanks.